examples/cpp/benchmark/PublisherApp.cpp

```diff
@@ -0,0 +1,545 @@
+                {
+                    std::cout << "Sample with index: '" <<
+                        benchmark_medium_.index() << "' (" << static_cast<int>(benchmark_medium_.data().size()) <<
+                        " Bytes) RECEIVED" << std::endl;
+                    sent = 0;
+                    if (elapsed.count() >= end_)
+                    {
+                        cv_.notify_one();
+                        return;
+                    }
+                    if (benchmark_medium_.index() > count)
+                    {
+                        benchmark_medium_.index(count);
+                    }
+                    else
+                    {
+                        benchmark_medium_.index(benchmark_medium_.index() + 1);
+                    }
+
+                    count = benchmark_medium_.index() + 1;
+                    if ((RETCODE_OK == writer_->write(&benchmark_medium_)) == true)
+                    {
+                        sent = 1;
+                        std::cout << "Sample with index: '" <<
+                            benchmark_medium_.index() << "' (" << static_cast<int>(benchmark_medium_.data().size()) <<
+                            " Bytes) SENT" << std::endl;
+                    }
+                }
+            }
+            break;
+
+        case CLIParser::MsgSizeKind::BIG:
+            while ((!is_stopped()) && (RETCODE_OK == reader->take_next_sample(&benchmark_big_, &info)))
+            {
+                if ((info.instance_state == ALIVE_INSTANCE_STATE) && info.valid_data)
+                {
+                    std::cout << "Sample with index: '" <<
+                        benchmark_big_.index() << "' (" << static_cast<int>(benchmark_big_.data().size()) <<
+                        " Bytes) RECEIVED" << std::endl;
+                    sent = 0;
+                    if (elapsed.count() >= end_)
+                    {
+                        cv_.notify_one();
+                        return;
+                    }
+                    if (benchmark_big_.index() > count)
+                    {
+                        benchmark_big_.index(count);
+                    }
+                    else
+                    {
+                        benchmark_big_.index(benchmark_big_.index() + 1);
+                    }
+
+                    count = benchmark_big_.index() + 1;
+                    if ((RETCODE_OK == writer_->write(&benchmark_big_)) == true)
+                    {
+                        sent = 1;
+                        std::cout << "Sample with index: '" <<
+                            benchmark_big_.index() << "' (" << static_cast<int>(benchmark_big_.data().size()) <<
+                            " Bytes) SENT" << std::endl;
+                    }
+                }
+            }
+            break;
+
+        default:
+            throw std::runtime_error("Type invalid");
+    }
+}
+
+void PublisherApp::run()
```

Comment:
I think we could simplify this method. 
1. Wait for the matches (2) or is_stopped() event (loop is not needed, just the matched_lock condition variable)
2. if we are not stopped: 
  3. Publish the corresponding message (this could be done in another method), 
  4. Do the initial wait
  5. Keep the same loop of lines 412:428.  
  6. Present results
 
What do you think ?